documentation/decisions/01 - Using a Compound Component Pattern for Autocomplete.md

```diff
@@ -0,0 +1,38 @@
+# Using a compound component pattern to build Autocomplete
+
+## Date
+
+October 3, 2019
+
+## Contributors
+
+- Andrew Musgrave
+- Chloe Rice
+- Dan Rosenthal
+- Daniel Leroux
+- Dominic McPhee
+- Sara Hill
+
+## Summary
+
+There is a need for Polaris to provide smaller, more composable components so that Polaris can be extended without needing to fork components. The re-development of the Autocomplete component provided an opportunity for exploration of the compound component pattern.
+
+## Solution
+
+Following an [exploration](https://github.com/Shopify/polaris-react/issues/1794), we decided to build the Autocomplete using a compound component pattern. We explored various ways to do this.
+
+1. Having the child delegate rendering to its parent via methods shared through context
+
+We opted against this for a couple of reasons:
+- it defeated the purpose of having smaller components
+- it causes the component to re-render every time an option is added (one call to the parent render method for each option)
```

Comment:
I wonder if we can go into more detail with this approach? It'll vary depending on architecture and complexity. A child `re-render` won't necessarily cause siblings to re-render. Here's a [sandbox example](https://codesandbox.io/s/amazing-river-6lomm)